nearcore/src/runtime/mod.rs

```diff
@@ -1408,14 +1438,18 @@ impl RuntimeAdapter for NightshadeRuntime {
         part_id: u64,
         num_parts: u64,
         data: &[u8],
+        epoch_id: &EpochId,
     ) -> Result<(), Error> {
         let part = BorshDeserialize::try_from_slice(data)
             .expect("Part was already validated earlier, so could never fail here");
-        let trie_changes = Trie::apply_state_part(&state_root, part_id, num_parts, part)
-            .expect("combine_state_parts is guaranteed to succeed when each part is valid");
+        let (trie_changes, contract_codes) =
+            Trie::apply_state_part(&state_root, part_id, num_parts, part)
+                .expect("combine_state_parts is guaranteed to succeed when each part is valid");
         let tries = self.get_tries();
         let (store_update, _) =
             tries.apply_all(&trie_changes, shard_id).expect("TrieChanges::into never fails");
+        let protocol_version = self.get_epoch_protocol_version(epoch_id)?;
```

Comment:
Is this the only reason we propagate epoch_id? If yes, why not just propogate epoch version instead?